### ISSUE TITLE
Missing foreign key between subscription and workflow

### ISSUE THREAD

**Issue by germandb** [danielgerlag/workflow-core#100]
Hi Daniell, 
I was deleting some old workflow from the database and realize that the subscription don't get deleted.
I saw that the subscription have a WorkflowId string that correspond with the workflow InstanceId, but have different types one its a string and the other a GUID and dont have a relationship.

**Comment by danielgerlag** [comment]
Yes, I made it loose on purpose... I'm in 2 minds about whether a constraint should be enforced due to the nature of the data.  I've also toyed with the idea of having a house keeping job to clean the subscriptions table of previously fired events.
